font/src/lib.rs

```diff
@@ -135,13 +142,46 @@ impl FontKey {
     }
 }
 
+#[cfg(feature = "hb-ft")]
+pub mod key_type {
+    #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+    pub enum KeyType {
+        // Harfbuzz returned a valid index from shaping and we can render that as expected.
+        GlyphIndex(u32),
+        // Harfbuzz returned nodef so we provide character instead to make use of font loading.
+        Fallback(char),
+    }
+    // Import enum variants into this module
+    use KeyType::{Fallback, GlyphIndex};
+
+    impl From<u32> for KeyType {
+        fn from(val: u32) -> Self {
+            GlyphIndex(val)
+        }
+    }
+    impl From<char> for KeyType {
+        fn from(val: char) -> Self {
+            Fallback(val)
+        }
+    }
+}
+#[cfg(feature = "hb-ft")]
+use key_type::KeyType;
+
+// For now use derived impls, this can be swapped for more efficient implementations once things are
+// working
```

Comment:
Things are mostly working, right? Should this actually be implemented manually?

If this is not performance relevant, I'm fine with leaving it derived and removing the comment to shorten the code.

Otherwise this should be implemented before merging.